src/Drupal/DrupalExtension/Context/DrupalContext.php

```diff
@@ -183,32 +183,134 @@ public function getTableRow(Element $element, $search)
         throw new \Exception(sprintf('Failed to find a row containing "%s" on the page %s', $search, $this->getSession()->getCurrentUrl()));
     }
 
+  /**
+   * Retrieve table rows containing specified text from a given element.
+   *
+   * @param Element $element
+   *   \Behat\Mink\Element\Element object.
+   * @param string $search
+   *   The text to search for in the table row.
+   *
+   * @return array of \Behat\Mink\Element\NodeElement
+   *
+   * @throws \RuntimeException
+   */
+  public function getTableRows(Element $element, $search): array {
+    $rows = $element->findAll('css', 'tr');
+    if (empty($rows)) {
+      throw new \RuntimeException(sprintf('No rows found on the page %s', $this->getSession()->getCurrentUrl()));
+    }
+    $found_rows = [];
+    foreach ($rows as $row) {
+      if (str_contains($row->getText(), $search) !== false) {
+        $found_rows += $row;
+      }
+    }
```

Comment:
@kumikoono Let's use a guard clause:
```suggestion
    foreach ($rows ?: [] as $row) {
      if (str_contains($row->getText(), $search) === FALSE) {
        continue;
      }
      $found_rows += $row;
    }
```

Reply:
Do we need this even we have the following just before it?

`    if (empty($rows)) {
      throw new \RuntimeException(sprintf('No rows found on the page %s', $this->getSession()->getCurrentUrl()));
    }`

Reply:
@kumikoono Yes, it is good to have even though we are sure `$row` is NOT empty for the future developer experience such as searching the entire code.  Also, we should consider "what if someone adds a code between `if`-statement and `for`-statement in the future and it might not be safe?"